ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1287,6 +1323,19 @@ - (NSString *)localizedInvalidValueStringWithAnswerString:(NSString *)text {
     return string;
 }
 
+- (NSString*)stringForAnswer:(id)answer {
+    if ([self isAnswerValid:answer]) {
+        NSNumberFormatter *formatter = [self makeNumberFormatter];
+        if (self.unit && self.unit.length > 0) {
+            [formatter setPositiveSuffix: [NSString stringWithFormat:@" %@", self.unit]];
+            [formatter setNegativeSuffix:[NSString stringWithFormat:@" %@", self.unit]];
+        }
+        return [formatter stringFromNumber:answer];
+    } else {
+        return nil;
+    }
+}
```

Comment:
It makes sense to check for `isAnswerValid` here, but then should you also check in all other answer formats as well. For example, a text answer format with a maximum length or regex.